### PR TITLE
Fix metrics appending in prometheus collector

### PR DESCRIPTION
**Pull request by jsoriano**
Update method of `common.MapStr` doesn't update nested maps
so we were overwriting existing metrics. 

Fix fields definition to add wildcard so fields match.

Add testdata.

### CODE REVIEW

**Comment by jsoriano** [comment]
Not adding changelog because prometheus refactor hasn't been released yet.

File: metricbeat/docs/fields.asciidoc [hunk 1]
```diff
@@ -20707,7 +20707,7 @@ Stats scraped from a Prometheus endpoint.
 
 
 
-*`prometheus.labels`*::
+*`prometheus.labels.*`*::
```
Comment (ruflin) [inline metricbeat/docs/fields.asciidoc]: Glad you found the convention ;-)

**Comment by sayden** [comment]
> Looks like the test failure is related. @sayden and @odacremolbap hit a similar issue on Windows with their prometheus changes, so I think it's more related to the test framework then the change itself.
> 
> @sayden @odacremolbap Has one of you found out the root cause of this?

I couldn't spent time on this yes but I was planning to start over again today. I have all the setup already done in Win to try

**Comment by jsoriano** [comment]
jenkins, test this again

**Comment by jsoriano** [comment]
I have renamed the file of the test that was failing in Windows and it passes now :unamused: Maybe something with the path lenght?

**Comment by odacremolbap** [comment]
LGTM,
waiting for @ruflin since I lack background on this fix

File: metricbeat/module/prometheus/collector/_meta/data.json [hunk 2]
```diff
@@ -14,15 +10,15 @@
     },
     "prometheus": {
         "labels": {
-            "interval": "15s"
+            "listener_name": "http"
         },
         "metrics": {
-            "prometheus_target_interval_length_seconds_count": 1,
-            "prometheus_target_interval_length_seconds_sum": 15.000226176
+            "net_conntrack_listener_conn_accepted_total": 3,
```
Comment (sayden) [inline metricbeat/module/prometheus/collector/_meta/data.json]: Is this change related?

Comment (jsoriano) [reply metricbeat/module/prometheus/collector/_meta/data.json]: Yes, this is generated from the `testdata/docs.json` file. This reproduced the issue this change is fixing as they are two different metrics with the same labels.
Previous example didn't reproduce the issue because they were values of the same summary or histogram.

**Comment by DanRoscigno** [comment]
Thanks Jaime!

File: metricbeat/docs/fields.asciidoc (hunk 1 above)
Comment (exekias) [reply metricbeat/docs/fields.asciidoc]: why is this a convention? I think this is the only `type: object` with `object_type: keyword` that uses `*`, example: https://github.com/elastic/beats/blob/master/libbeat/_meta/fields.ecs.yml#L23

Comment (jsoriano) [reply metricbeat/docs/fields.asciidoc]: We investigated this offline, it seems that both options generate equivalent dynamic templates (`.*` is appended if there is no `*`). But the new fields validator for the results of testdata requires the `*`.

Comment (ruflin) [reply metricbeat/docs/fields.asciidoc]: Jaime is correct, it's for the new test framework.
